src/background/responders.ts

```diff
@@ -130,5 +93,105 @@ export const responders: { [T in Action['type']]: Responder<T> } = {
       justTweeted: null,
       alert: 'POST TWEET FAILURE'
     }
+  },
+  'chrome.windows.getAll'(state, action) {
+    return {
+      focusedWindowId: action.payload.windows.find(win => win.focused)!.id
+    }
+  },
+  'chrome.tabs.query'(state, action) {
+    const tabs: AppState['tabs'] = new Map()
+    action.payload.tabs.forEach(tab =>
+      tabs.set(tab.id!, {
```

Comment:
According to the docs, tab IDs can be undefined. Also the same for all the rest of the attributes we `!`

Reply:
Since we're not [querying foreign tabs using the sessions API](https://developer.chrome.com/extensions/tabs#type-Tab), we should be fine